### ISSUE TITLE
A11y issues on Courses page in 0.19.2-alpha0

### ISSUE THREAD

**Issue by radinamatic** [learningequality/kolibri#14196]
<img height="20px" src="https://i.imgur.com/c7hUeb5.jpeg">

❌ **This issue is not open for contribution. Visit <a href="https://learningequality.org/contributing-to-our-open-code-base/" target="_blank">Contributing guidelines</a>** to learn about the contributing process and how to find suitable issues.

<img height="20px" src="https://i.imgur.com/c7hUeb5.jpeg">

## Observed behavior
1. `Select` drop-downs for status and recipients are not keyboard navigable (dropdown opens but options under it cannot be selected). Not a regression, and there are probably already KDS issues filed for that, just a reminder that fixing that is still pending.

   <img width="657" height="455" alt="Image" src="https://github.com/user-attachments/assets/68ecc50c-ae5c-4def-aaed-13e19d273060" />

2. `aria-label` cannot be used on generic elements like `span`, which do not have a role, and will be ignored by the screen readers. Since the element is not interactive and is visually within the *Search* input field that has its own label, it does not add much to the user experience if we tried to add a role to it or switch to `visuallyhidden` presentation, so let's just remove it.
   
   <img width="1594" height="456" alt="Image" src="https://github.com/user-attachments/assets/344bc24e-3f46-46a9-a844-e2c23ff9336a" />

3. Are there plans to use the accessible `KTable` component here? Independent of that, what will probably add considerable cognitive overload for the screen reader users is the fact that both the course title AND description are included in the first cell of the rows. Not sure how much benefit that carries even for the sighted users since it's going to be cut at the column edge, but at the very least I would suggest to hide it for screen reader users with 

   `<span data-v-...="" class="course-description" aria-hidden="true"...` 

   <img width="1648" height="458" alt="Image" src="https://github.com/user-attachments/assets/52eb10e2-a170-4e0c-8cad-30e3f92e3120" />

4. None of the toggles for course visibility nor buttons for course options have labels. While the ones for the latter should be obvious (`Options`) and easy to add to the button, for the toggles it might require more complex options... As of now the NVDA output is only `checkbox not checked`, so clearly not informative enough. I'm also wondering why the `input` element  does not have the `role="switch"`...? 🤔 

   <img width="1675" height="639" alt="Image" src="https://github.com/user-attachments/assets/a8fe167f-92a8-4cdf-94cf-92201e33916f" />

5. Another thing that is missing is the proper implementation of the `aria-live` to receive and announce the snack-bar notification that appears after the visibility is changed.

   <img width="1563" height="998" alt="Image" src="https://github.com/user-attachments/assets/6fe42fd2-cb94-4944-a2c1-13c2544f329d" />

6. Course cards should be heading level 2, not level 3, since the sidebar title is level 1.

   <img width="1363" height="1059" alt="Image" src="https://github.com/user-attachments/assets/0313aa41-70a4-49ff-9ad8-0e2bf8c68ccf" />

7. It's not possible to activate the radio button *Entire class* by keyboard (neither `Space` nor `Enter` manage to do the job).

   <img width="1202" height="487" alt="Image" src="https://github.com/user-attachments/assets/30a25aae-29b3-45e9-905f-12e0b5c7df6e" />

8. Course units buttons are missing the visible focus outline: the one in focus in the screenshot below is the *Feeding & Metabolism* but is visually indistinguishable from the rest.

   <img width="1112" height="968" alt="Image" src="https://github.com/user-attachments/assets/741ed26c-3fe5-4aed-a524-97827e98a8ed" />

## Errors and logs
…

## Expected behavior
…

## User-facing consequences
…

## Steps to reproduce
…

## Context
 * Application version: 0.19.2-alpha0
 * Operating system: Windows 10
 * Browser: FF & Chrome with NVDA

**Comment by MisRob** [comment]
> Another thing that is missing is the proper implementation of the aria-live to receive and announce the snack-bar notification that appears after the visibility is changed.

Few organizational & technical notes.

This is a global problem. Will be resolved with the new `KSnackbar` component :) https://github.com/learningequality/studio/issues/5445. I think it'd be best to wait for that.

Or, if it feels urgent, let's please use https://design-system.learningequality.org/usekliveregion (versus just adding another `aria-live` that we would need to remove https://github.com/learningequality/kolibri/issues/12326)

**Comment by marcellamaki** [comment]
Closed as mostly-completed in #14645 with follow up required, tracked in: https://github.com/learningequality/kolibri-design-system/issues/1248, https://github.com/learningequality/studio/issues/5445
